redbot/core/core_commands.py

```diff
@@ -1435,21 +1436,27 @@ async def traceback(self, ctx: commands.Context, public: bool = False):
         **Arguments:**
             - `[public]` - Whether to send the traceback to the current context. Leave blank to send to your DMs.
         """
-        if not public:
-            destination = ctx.author
-        else:
-            destination = ctx.channel
+
+        copied_ctx = copy(ctx)
+        copied_ctx.channel = (
+            ctx.channel
+            if public
+            else (
+                ctx.author.dm_channel if ctx.author.dm_channel else await ctx.author._get_channel()
+            )
+        )
```

Comment:
I get the purpose of this but this is extremely hacky for what it's used for... If we *have to* override a context's channel I think we could do away without a copy but I feel like we shouldn't have to. Maybe it's time to have a `send_interactive` utility that isn't tied to a Context object :thinking: